src/mpid/ch3/src/ch3u_request.c

```diff
@@ -200,14 +200,15 @@ int MPIDI_CH3U_Request_load_send_iov(MPIR_Request * const sreq,
 int MPIDI_CH3U_Request_load_recv_iov(MPIR_Request * const rreq)
 {
     MPI_Aint last;
-    static intptr_t orig_segment_first = MPIDI_LOAD_RECV_IOV_ORIG_SEGMENT_FIRST_UNSET;
+    if (rreq->dev.segment_first == 0)
```

Comment:
in case if .segment_first in non-zero what is default value of .orig_segment_first?

what about add setting of default value for .orig_segment_first to MPID_Request_create?